internals/overlord/checkstate/manager.go

```diff
@@ -633,3 +644,50 @@ func (m *CheckManager) Replan() {
 		m.updateCheckData(check, changeID, 0)
 	}
 }
+
+// RefreshCheck runs a check immediately.
+func (m *CheckManager) RefreshCheck(ctx context.Context, check *plan.Check) (*CheckInfo, error) {
+	m.checksLock.Lock()
+	checkData := m.ensureCheck(check.Name)
```

Comment:
Nit: let's rename this local variable to just `data` throughout. This is all in the context of "check", and it would avoid shadowing the type name.